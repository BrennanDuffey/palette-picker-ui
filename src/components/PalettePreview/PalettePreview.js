import React from 'react';

function PalettePreview(props) {
  const colors = props.colors;
  const colorPreviews = colors.map(color => 
    <div 
      className="preview"
      style={{ background: color.hexCode}}
      key={color.id}>
    </div>
  );

  return (
    <article className="PalettePreview">
      <h4>Palette Name</h4>
      <div className="flex-container">
        {colorPreviews}
        <i 
          onClick={() => props.handleDeletePalette(props.paletteId)}
          className="fas fa-times">
        </i>
      </div>
    </article>
  );
}

export default PalettePreview;