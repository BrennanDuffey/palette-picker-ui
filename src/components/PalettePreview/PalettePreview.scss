.PalettePreview {
  h4 {
    font-weight: 300;
    margin: 30px 0 5px;
  }

  .preview {
    background: $lightGray;
    height: 50px;
    width: 50px;
  }
}

.flex-container {
  display: flex;
}