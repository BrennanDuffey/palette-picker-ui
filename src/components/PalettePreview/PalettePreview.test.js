import React from 'react';
import { shallow } from 'enzyme';
import PalettePreview from './PalettePreview';

describe('PalettePreview', () => {
  let mockColors;
  let wrapper;

  beforeEach(() => {
    mockColors = [
      '#FDC58D', 
      '#FFFFAA', 
      '#95E38A', 
      '#779DF2', 
      '#D2BEFD'
    ]
    wrapper = shallow(
      <PalettePreview
        colors={mockColors}
      />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});