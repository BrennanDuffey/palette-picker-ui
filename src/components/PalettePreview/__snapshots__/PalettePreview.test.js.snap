// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PalettePreview should match snapshot 1`] = `
<article
  className="PalettePreview"
>
  <h4>
    Palette Name
  </h4>
  <div
    className="flex-container"
  >
    <div
      className="preview"
      key="1562544710381"
      style={
        Object {
          "background": "#FDC58D",
        }
      }
    />
    <div
      className="preview"
      key="1562544710381"
      style={
        Object {
          "background": "#FFFFAA",
        }
      }
    />
    <div
      className="preview"
      key="1562544710381"
      style={
        Object {
          "background": "#95E38A",
        }
      }
    />
    <div
      className="preview"
      key="1562544710381"
      style={
        Object {
          "background": "#779DF2",
        }
      }
    />
    <div
      className="preview"
      key="1562544710381"
      style={
        Object {
          "background": "#D2BEFD",
        }
      }
    />
  </div>
</article>
`;
